ch_tools/chadmin/cli/object_storage_group.py

```diff
@@ -129,146 +74,180 @@ def object_storage_group(ctx: Context, config_path: Path, disk_name: str) -> Non
+
+    remote_data_paths_table = "system.remote_data_paths"
+    if on_cluster:
+        remote_data_paths_table = (
+            f"clusterAllReplicas('{cluster_name}', {remote_data_paths_table})"
+        )
 
-            if orphaned and metadata:
-                continue
-            if not orphaned and not metadata:
-                continue
+    antijoin_query = f"""
+        SELECT obj_path FROM {listing_table} AS object_storage
+          LEFT ANTI JOIN {remote_data_paths_table} AS object_table
+          ON object_table.remote_path = object_storage.obj_path
+            AND object_table.disk_name = '{disk_conf.name}'
+    """
+    logging.info("Antijoin query: %s", antijoin_query)
 
-            writer.write(_get_dump_line(obj, metadata, quiet))
-            counter += 1
+    if dry_run:
+        click.echo("Counting orphaned objects...")
+    else:
+        click.echo("Deleting orphaned objects...")
+
+    deleted = 0
+    with execute_query(
+        ctx, antijoin_query, stream=True, format_="TabSeparated"
+    ) as resp:
+        # make generator for lazy iterating
+        paths_to_delete = (line.decode() for line in resp.iter_lines())
+        deleted = cleanup_s3_object_storage(disk_conf, paths_to_delete, dry_run)
+
+    click.echo(
+        f"{'Would delete' if dry_run else 'Deleted'} {deleted} objects from bucket [{disk_conf.bucket_name}] with prefix {disk_conf.prefix}"
+    )
 
 
-@object_storage_group.command("clean")
-@option(
-    "--prefix",
-    "prefix",
-    help="Prefix path to delete objects. If not empty it has more priority than other options.",
-)
-@option(
-    "-f",
-    "--file",
-    "file",
-    type=click.File("rb"),
-    help="File containing S3 object keys delimited by newlines.",
-    default=sys.stdin.buffer,
-    show_default="STDIN",
-)
-@option(
-    "-c",
-    "--compressed",
-    "compressed",
-    is_flag=True,
-    help="Input stream is compressed using GZIP format.",
-)
-@pass_context
-def clean_object_storage(ctx, prefix, file, compressed):
+def _traverse_object_storage(
+    ctx: Context,
+    listing_table: str,
+    from_time: Optional[timedelta],
+    to_time: timedelta,
+    prefix: str,
+) -> None:
     """
-    Clean up needless S3 objects.
+    Traverse S3 disk's bucket and put object names to the ClickHouse table.
     """
-    disk_conf: S3DiskConfiguration = ctx.obj["disk_configuration"]
-
-    if compressed:
-        file = GzipFile(fileobj=file)
-    file = TextIOWrapper(file)
+    obj_paths_batch = []
+    counter = 0
+    now = datetime.now(timezone.utc)
 
-    if prefix:
-        file = (
-            obj.key
-            for _, obj in s3_object_storage_iterator(
-                disk_conf, path_prefix=prefix, skip_ignoring=True
-            )
-        )
+    for _, obj in s3_object_storage_iterator(
+        ctx.obj["disk_configuration"], object_name_prefix=prefix
+    ):
+        if obj.last_modified > now - to_time:
```

Comment:
In the knobs description:
```
"Objects with a modification time falling interval [now - from_time, now - to_time] are considered.
```

But here `now-to_time` is the lower bound.

Reply:
It is an upper bound of an interval
https://github.com/yandex/ch-tools/blob/3dfdae2f8f0d5540ae202a964cb8572496dc9cca/ch_tools/chadmin/cli/object_storage_group.py#L238-L239

Actually we filter objects newer than `now - 24h` point in time and pass objects created more than 24h ago.